data_transformation.py

```diff
@@ -1,21 +1,134 @@
+    text = remove_emojis(text)                                      # Removes all Emojis #TODO If we use a HF model we don't need to remove emojis
+    text = re.sub(r"\b[a-zA-Z]\b", "", str(text))                   # Removes all single characters
+    text = re.sub(r"[^\w\s]", " ", str(text))                       # Removes all punctuations
+    text = re.sub(r'(.)\1+', r'\1\1', text)                         # Convert more than 2 letter repetitions to 2
+    text = re.sub(r"\s+", " ", str(text))                           # Replaces double spaces with single space    
+    return text
+
+
+def count_emoji(text):
+    positive = len(re.findall(r'positive_emoji', text))
+    negative = len(re.findall(r'negative_emoji', text))
+    return {'positive': positive, 'negative':negative}
 
 
 def transform_twitter_data(data):
-    return data
+    data.rename(columns={'text': 'original_text'}, inplace=True)
+    data['processed_text'] = data['original_text'].apply(lambda x: process_data(x))
+    data['emojis'] = data['processed_text'].apply(lambda x: count_emoji(x))
+
+
+def get_sentiments(data):
+    sentiment_pipeline = pipeline(model=MODEL_PATH)
 
+    data['output_clean'] = data['processed_text'].apply(lambda tweet: sentiment_pipeline(tweet))
+    data['sentiment_label'] = data['output_clean'].apply(lambda dic: dic[0]['label'])
+    data['sentiment_score'] = data['output_clean'].apply(lambda dic: dic[0]['score'])
+    #TODO drop output_clean?
 
-def calculate_company_index(linkedin_data, twitter_data, sentiments_df):
+
+def calculate_company_index(data):
     # TODO we should create kind of a formula to calculate company_index
+    # should we consider retweets, marked as favorite?
     company_index = 0
     return company_index
 
 
-def get_sentiments(data):
-    sentiments = pd.DataFrame()
-    return sentiments
+def count_words(text, stop_words):
+    words_list = text.split()
+    words_list = [word for word in words_list if not word in stop_words]
+    freq = [words_list.count(w) for w in words_list]
+    return dict(list(zip(words_list,freq)))
+
+
+def count_words_all_texts(serie_text, stop_words):
+    text = serie_text.to_list()
+    text = ' '.join(text)
+    return count_words(text, stop_words)
+
+def sort_df_freq_words(serie_text, stop_words):
+    dict_word = count_words_all_texts(serie_text, stop_words)
+    df = pd.DataFrame(dict_word.items(), columns=['Word', 'Freq'])
+    return df.sort_values('Freq',ascending=False)
+
+
+def get_ranked_tweets(data, stop_words, filter_condition):
+    tweets = data['processed_text'][data['sentiment_label']==filter_condition]
+    df_tw = sort_df_freq_words(tweets, stop_words)
+    df_tw = df_tw[0:10] #FIXME how many should we show?
+    df_tw['positive'] = True if filter_condition=='positive' else False
+    return df_tw
+
+
+def get_top_pos_neg_tokens(company_name, data):
+    # FIXME from where "https", "co", "RT" came? should we remove them when cleaning data?
```

Comment:
this is for you :)  @RomiLanda

Reply:
I am going to check out "co" and "RT", but in the **process _data**  we remove HTTPS with this code: `text = re.sub('((www\.[^\s]+)|(https?://[^\s]+))', ' ', text)`.

Reply:
k, I'll leave that comment so you can work on it when ypu find time